GOAT-Compiler/Code Generation/CodeGenerator.cs

```diff
@@ -0,0 +1,961 @@
+            {
+                node.GetBlock().Apply(this);
+            }
+            OutAProcDecl(node);
+        }
+
+
+        public override void InsideScopeOutAProcDecl(AProcDecl node)
+        {
+            CallStackRT.Pop();
+        }
+
+        public override void CaseAGcodeStmt(AGcodeStmt node)
+        {
+            string literal = node.GetGcodeLiteral().Text;
+            var linebyline = literal.Split(new string[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+            foreach (string line in linebyline)
+            {
+                _textWriter.WriteLine(line.Trim());
+            }
+        }
+
+
+        internal class RuntimeTable<TKey>
+        {
+            private Dictionary<TKey, int> IntMap = new Dictionary<TKey, int>();
+            private Dictionary<TKey, double> FloatMap = new Dictionary<TKey, double>();
+            private Dictionary<TKey, bool> BoolMap = new Dictionary<TKey, bool>();
+            private Dictionary<TKey, Vector> VecMap = new Dictionary<TKey, Vector>();
+
+            public void Put(TKey key, int value)
+            {
+                if (IntMap.ContainsKey(key))
+                {
+                    IntMap[key] = value;
+                }
+                else
+                {
+                    IntMap.Add(key, value);
+                }
+            }
+
+            public void Put(TKey key, bool value)
+            {
+                if (BoolMap.ContainsKey(key))
+                {
+                    BoolMap[key] = value;
+                }
+                else
+                {
+                    BoolMap.Add(key, value);
+                }
+            }
+
+            public void Put(TKey key, double value)
+            {
+                if (FloatMap.ContainsKey(key))
+                {
+                    FloatMap[key] = value;
+                }
+                else
+                {
+                    FloatMap.Add(key, value);
+                }
+            }
+
+            public void Put(TKey key, Vector vector)
+            {
+                if (VecMap.ContainsKey(key))
+                {
+                    VecMap[key] = vector;
+                }
+                else
+                {
+                    VecMap.Add(key, vector);
+                }
+            }
+
+            public dynamic Get(TKey key, Types type)
+            {
+                switch (type)
+                {
+                    case Types.Integer:
+                        return IntMap[key];
+                    case Types.FloatingPoint:
+                        return FloatMap[key];
+                    case Types.Boolean:
+                        return BoolMap[key];
+                    case Types.Vector:
+                        return VecMap[key];
+                    default:
+                        return null;
+                }
+            }
+
+            public bool Contains(TKey key)
+            {
+                if (IntMap.ContainsKey(key) ||
```

Comment:
the if is redundant, just return the expression.
If (bool-exp){
return true;
}
EQUALS:
return bool-exp;

Reply:
fixed